Assets/LeapMotion/Scripts/HandPool.cs

```diff
@@ -59,35 +84,63 @@ public class ModelGroup {
      * @param hand The Leap Hand data to be drive an IHandModel
      * @param modelType Filters for a type of hand model, for example, physics or graphics hands.
      */
+
     public override HandRepresentation MakeHandRepresentation(Hand hand, ModelType modelType) {
-      HandRepresentation handRep = null;
+      //HandRepresentation handRep = null;
       List<IHandModel> models = new List<IHandModel>();
-      foreach (ModelGroup group in ModelPool) {
-        for (int i = 0; i < group.modelList.Count; i++) {
-          IHandModel model = group.modelList[i];
-          bool isCorrectHandedness;
-          Chirality handChirality = hand.IsRight ? Chirality.Right : Chirality.Left;
-          isCorrectHandedness = model.Handedness == handChirality;
-          if (!EnforceHandedness || model.Handedness == Chirality.Either) {
-            isCorrectHandedness = true;
-          }
-          bool isCorrectModelType;
-          isCorrectModelType = model.HandModelType == modelType;
-          if (isCorrectModelType && isCorrectHandedness) {
-            group.modelList.RemoveAt(i);
-            //--i;
-            models.Add(model);
-            break;
+      Chirality handChirality = hand.IsRight ? Chirality.Right : Chirality.Left;
+      HandRepresentation handRep = new HandProxy(this, hand, handChirality, modelType);
+      for (int i = 0; i < ModelPool.Count; i++) {
+        ModelGroup group = ModelPool[i];
+        if (group.IsEnabled) {
+          IHandModel model = group.TryGetModel(handChirality, modelType);
+          if (model != null) {
+            handRep.AddModel(model);
+            group.modelsCheckedOut.Add(model);
```

Comment:
Instead of forcing HandPool to manually take care of the logic when checking out models, TryGetModel should handle this logic automatically to prevent error.  There might be a matching ReturnModel method as well that automatically removes it from the modelsCheckOut list.